wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -220,156 +173,115 @@ class PresentationActivity : FragmentActivity() {
-                        horizontalAlignment = Alignment.CenterHorizontally
-                    ) {
-                        Text(
-                            text = "Sending mDL to reader.",
-                            style = MaterialTheme.typography.bodyMedium,
-                            color = MaterialTheme.colorScheme.onSurface
-                        )
-                        Text(
-                            text = "TODO: finalize UI",
-                            style = MaterialTheme.typography.bodyMedium,
-                            color = MaterialTheme.colorScheme.onSurface
-                        )
-                        Divider()
-                        Text(
-                            text = "State: ${stateDisplay.value}",
-                            style = MaterialTheme.typography.bodyMedium,
-                            color = MaterialTheme.colorScheme.onSurface
-                        )
-                        Divider()
-                        Button(onClick = { finish() }) {
-                            Text("Close")
-                        }
-                    }
-
-                    // when consent data is available, show the consent prompt above the activity's UI
-                    consentData = consentPromptData.value
-                    if (consentData != null) {
-                        ConsentPromptEntryField(
-                            consentData = consentData!!,
-                            documentTypeRepository = walletApp.documentTypeRepository,
-                            onConfirm = { // user accepted to send requested credential data
-                                finishProcessingRequest()
-                            },
-                            onCancel = { // user declined submitting data to requesting party
-                                finish() // close activity
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = Color.Transparent
+                ) {
+                    // if Presentation data is parsed and ready to use, show the Presentation Flow
+                    parsedPresentationRequestData.value?.let { presentationRequestData ->
+                        // launched effect [key1] does not change for presentations,
+                        // either the flow succeeds or Activity is finish()'ed
+                        LaunchedEffect(key1 = "MDL_Presentation") {
+
+                            try {
+                                val responseBytes = showPresentationFlow(presentationRequestData)
+                                sendResponseToDevice(responseBytes)
+                            } catch (exception: Exception) {
+                                Logger.e(TAG, "Error finishing the Presentation flow: $exception")
                             }
-                        )
+                        }
                     }
                 }
             }
         }
     }
```

Comment:
I think onCreate() could be a lot simpler
- need for the Compose thing at all (setContent {)
- do away with handleActivityStateChanges(), just incorporate it into onCreate()